pkg/landscaper/controllers/deployitem/controller.go

```diff
@@ -166,8 +165,10 @@ func (con *controller) reconcileOld(ctx context.Context, req reconcile.Request)
 	return reconcile.Result{RequeueAfter: *requeue}, nil
 }
 
-func (con *controller) detectPickupTimeouts(log logging.Logger, di *lsv1alpha1.DeployItem) (*time.Duration, error) {
-	logger := log.WithValues("operation", "DetectPickupTimeouts")
+func (con *controller) detectPickupTimeouts(ctx context.Context, di *lsv1alpha1.DeployItem) (*time.Duration, error) {
+	logger, _ := logging.FromContextOrNew(ctx, nil)
+	logger = logger.WithValues(lc.KeyMethod, "detectPickupTimeouts")
```

Comment:
```suggestion
	logger, ctx := logging.FromContextOrNew(ctx, nil, lc.KeyMethod, "detectPickupTimeouts")
```

Reply:
I want to use the logging key `lc.KeyMethod` only in the present method, so that sub methods can add their own method name to the logger. Therefore, I do not add loggers with a "local" key like `lc.KeyMethod` to the context.

Reply:
Ah, I see. Valid point, adding the method name should only be done if there aren't any interesting submethods anymore.